app/src/main/java/ch/epfl/polychef/OnlineMiniaturesFragment.java

```diff
@@ -0,0 +1,95 @@
+package ch.epfl.polychef;
+
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.Fragment;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+import ch.epfl.polychef.adaptersrecyclerview.RecipeMiniatureAdapter;
+import ch.epfl.polychef.recipe.Recipe;
+
+public class OnlineMiniaturesFragment extends Fragment implements FireHandler {
+
+    private RecyclerView onlineRecyclerView;
+
+    private List<Recipe> dynamicRecipeList = new ArrayList<>();
+
+    private int currentReadInt = 1;
+
+    private int nbOfRecipesLoadedAtATime = 5;
+
+    private boolean isLoading = false;
+
+    private Firebase firebase;
+
+    public OnlineMiniaturesFragment(){
+        
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState){
+        View view = inflater.inflate(R.layout.fragment_miniatures_online, container, false);
+
+        Bundle bundle = getArguments();
+        int fragmentID = bundle.getInt("fragmentID");
+        onlineRecyclerView = view.findViewById(R.id.miniaturesOnlineList);
+        onlineRecyclerView.setLayoutManager(new LinearLayoutManager(this.getActivity()));
+        onlineRecyclerView.setAdapter(new RecipeMiniatureAdapter(this.getActivity(), dynamicRecipeList, onlineRecyclerView, fragmentID));
+        // Add a scroll listener when we reach the end of the list we load new recipes from database
+        onlineRecyclerView.addOnScrollListener(new RecyclerView.OnScrollListener() {
+            @Override
+            public void onScrollStateChanged(@NonNull RecyclerView recyclerView, int newState) {
+                super.onScrollStateChanged(recyclerView, newState);
+
+                if(!recyclerView.canScrollVertically(1)){
+                    if(isLoading){
+                        return;
+                    }
+                    // add a certain number of recipes at this end of the actual list
+                    for(int i = 0; i < nbOfRecipesLoadedAtATime; i++){
+                        Firebase.readRecipeFromFirebase(currentReadInt, OnlineMiniaturesFragment.this);
```

Comment:
Should make a `loadNRecipes(int n)` method so that we don't need to make `n` queries every time.